ldap_auth_provider.py

```diff
@@ -447,8 +449,6 @@ async def _fetch_root_domain(self) -> str:
         )
 
         if result:
```

Comment:
I'd suggest an early return here, to make it easy to follow:

```suggestion
        if not result:
            logger.warning("Unable to get root domain")
            return self.ldap_root_domain
            
        # ... etc
```

also: the warning message probably needs improving. *why* were we unable to get the root domain?

Reply:
> I'd suggest an early return here, to make it easy to follow:

There will be 2 `return self.ldap_root_domain` statements then. I personally don't like such duplication, but ok.

Reply:
I find early returns much easier to follow than big `if` statements.

Reply:
https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement

Reply:
To be honest, neither example at the stackexchange produces duplicated code and starting question is about void function.  
But as stated at question header - overall question it's opinion based.  

I do early return always when it doesn't produced duplicated return statements.